lib/index.js

```diff
@@ -337,6 +343,42 @@ Server.prototype = {
       yield this.apiml.registerMainServerInstance();
     }
   }),
+  
+  configureApimlStorage(apimlConfig) {
+    const { gatewayHost, gatewayPort } = apimlConfig.server;
+    const { cachingService } = apimlConfig;
+    if (typeof gatewayHost !== 'string' || typeof gatewayPort !== 'number') {
+      return;
+    }
+    if (typeof cachingService !== 'object') {
+      return;
+    }
+    const {cert: clientCertFile, key: clientKeyFile} = cachingService;
+    if (!clientCertFile || !clientKeyFile) {
+      return;
+    }
+    let cert, key;
+    try {
+      cert = fs.readFileSync(clientCertFile);
+      key = fs.readFileSync(clientKeyFile);
+    } catch (e) {
+      bootstrapLogger.warn(`ZWED0176W`, e.String()) // Failed to load client cert/key pair for Caching Service (%s)
+      return;
+    }
+    if (key && cert && gatewayHost && gatewayPort) {
+      const settings = {
+        host: gatewayHost,
+        port: gatewayPort,
+        tlsOptions: {
+          cert: cert,
+          key: key,
+          rejectUnauthorized: false,
```

Comment:
can it ever be self signed? shouldnt this just use the tlsOptions from webserver.js?

Reply:
yes, you are right. see my previous comment. I fixed it.